src/js/modules/SelectRange/SelectRange.js

```diff
@@ -908,27 +910,29 @@ export default class SelectRange extends Module {
 		return range;
 	}
 	
-	resetRanges() {
+	resetRanges(forceNewRange = true) {
 		var range, cell, visibleCells;
-		
+
 		this.ranges.forEach((range) => range.destroy());
 		this.ranges = [];
-		
-		range = this.addRange();
-		
-		if(this.table.rowManager.activeRows.length){
-			visibleCells = this.table.rowManager.activeRows[0].cells.filter((cell) => cell.column.visible);
-			cell = visibleCells[this.rowHeader ? 1 : 0];
 
-			if(cell){
-				range.setBounds(cell);
-				if(this.options("selectableRangeAutoFocus")){
-					this.initializeFocus(cell);
+		if(forceNewRange || this.options("selectableRangeInitializeDefault")) {
+			range = this.addRange();
+
+			if(this.table.rowManager.activeRows.length){
+				visibleCells = this.table.rowManager.activeRows[0].cells.filter((cell) => cell.column.visible);
+				cell = visibleCells[this.rowHeader ? 1 : 0];
+
+				if(cell){
+					range.setBounds(cell);
+					if(this.options("selectableRangeAutoFocus")){
+						this.initializeFocus(cell);
+					}
```

Comment:
I think this whole thing could be a new method? We can call it `setInitialRange()` or something similar.

Reply:
I refactored `resetRanges` into `resetRanges` and `setDefaultRange`